app/controllers/SubmissionsController.scala

```diff
@@ -45,4 +49,15 @@ class SubmissionsController @Inject()(
     } yield Ok(submissionsPage(result.toSeq))
   }
 
+  def amend(id: String): Action[AnyContent] = authenticate.async { implicit request =>
+    customsDeclareExportsConnector.findDeclaration(authenticatedSessionId, id) flatMap {
```

Comment:
there is `sessionId` on request - we don't have to use `SessionIdAware` trait

Reply:
We probably dont need this trait any more at all